include/deal.II/grid/reference_cell.h

```diff
@@ -183,6 +183,77 @@ namespace ReferenceCell
     return Point<dim>(+0.0, +0.0, +0.0);
   }
 
+  /**
+   * Return i-th unit tangential vector (array of dim-1 elements of Point<dim>)
+   * of a face of the reference cell. The vectors are arranged such that the
+   * cross product between the two vectors return the unit normal vector.
+   */
+  template <int dim>
+  inline Tensor<1, dim>
+  unit_tangential_vectors(const Type &       reference_cell,
+                          const unsigned int face_no,
+                          const unsigned int i)
+  {
+    if (reference_cell == get_hypercube(dim))
+      return GeometryInfo<dim>::unit_tangential_vectors[face_no][i];
+    else if (reference_cell == Type::Tri)
+      {
+        AssertDimension(i, 0);
```

Comment:
The other cases don't seem to have this assertion. I think you could put it at the top of the function if you manage to extract the dimensionality of the `reference_cell` object somehow.

Reply:
This assert is only needed here since this is the only cell type that is 2D (and a consequence only has a single tangential).

Reply:
But what if someone calls the function with `reference_cell==wedge` and `i==14`? In that case, you're causing an out-of-bounds access for `std::array` that isn't caught.